src/grpc/retry-interceptor.ts

```diff
@@ -92,48 +62,60 @@ export class RetryInterceptor {
               // eslint-disable-next-line @typescript-eslint/no-explicit-any
               next: (arg0: any) => void
             ) {
-              let retries = 0;
+              let attempts = 1;
               const retry = function (message: unknown, metadata: Metadata) {
-                retries++;
+                attempts++;
                 const newCall = nextCall(options);
                 newCall.start(metadata, {
                   onReceiveMessage: function (message) {
                     savedReceiveMessage = message;
                   },
                   onReceiveStatus: function (status) {
-                    if (retryableGrpcStatusCodes.includes(status.code)) {
-                      if (retries <= maxRetry) {
-                        logger.debug(
-                          `Request path: ${options.method_definition.path}; retryable status code: ${status.code}; number of retries (${retries}) is less than max (${maxRetry}), retrying.`
-                        );
-                        retry(message, metadata);
-                      } else {
-                        logger.debug(
-                          `Request path: ${options.method_definition.path}; retryable status code: ${status.code}; number of retries (${retries}) has exceeded max (${maxRetry}), not retrying.`
-                        );
-                        savedMessageNext(savedReceiveMessage);
-                        next(status);
-                      }
-                    } else {
+                    const whenToRetry =
+                      retryStrategy.determineWhenToRetryRequest({
+                        grpcStatus: status,
+                        grpcRequest: options.method_definition,
+                        attemptNumber: attempts,
+                      });
+
+                    if (whenToRetry === null) {
+                      logger.debug(
+                        `Request not eligible for retry: path: ${options.method_definition.path}; retryable status code: ${status.code}; number of attempts (${attempts}).`
+                      );
                       savedMessageNext(savedReceiveMessage);
-                      next({code: status.code});
+                      next(status);
+                    } else {
+                      `Request eligible for retry: path: ${options.method_definition.path}; response status code: ${status.code}; number of attempts (${attempts}); will retry in ${whenToRetry}ms`;
```

Comment:
Does this do anything without a logger call?

Reply:
nope thanks :)

Reply:
fixed in https://github.com/momentohq/client-sdk-javascript/pull/205/commits/15e2d8ea91e9977d19ba25e87b865123dd5c8b28